### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by ssh24**
### Description

Use a dockerized container to run tests.

connect to https://github.com/strongloop-internal/scrum-apex/issues/271

**Please note:** The windows build for db2 is turned off due to this. The softlayer VM is hosted on an IBM network and the windows machine (hosted outside of IBM network) cannot connect to it.

### CODE REVIEW

**Comment by ssh24** [comment]
@slnode test please

File: test/describe.js [hunk 1]
```diff
@@ -1,3 +0,0 @@
-'use strict';
```
Comment (b-admike) [inline test/describe.js]: Is this file not needed anymore?

Comment (ssh24) [inline test/describe.js]: Nope. We can finally **unskip** the tests. :D

**Comment by ssh24** [comment]
@slnode test please

File: lib/discovery.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ function mixinDiscovery(DB2, db2) {
         ' TRIM(tabschema) AS "owner",' +
         ' TRIM(property) AS "property"' +
         ' FROM syscat.tables where substr(property,20,1) NOT LIKE \'Y\'',
-        'table_schema, table_name', options);
+      'table_schema, table_name', options);
```
Comment (dhmlau) [inline lib/discovery.js]: the original spacing seems to be better.  Any reason to change it?

File: lib/discovery.js [hunk 3]
```diff
@@ -79,15 +79,15 @@ function mixinDiscovery(DB2, db2) {
         ' FROM syscat.tables' +
         ' WHERE tabschema=\'' + schema + '\' AND' +
         ' SUBSTR(property, 20, 1) NOT LIKE \'Y\'',
-        'table_schema, table_name', options);
+      'table_schema, table_name', options);
```
Comment (dhmlau) [inline lib/discovery.js]: same question as above

File: test/db2.migration.test.js [hunk 4]
```diff
@@ -28,14 +26,14 @@ describe('migrations', function() {
...
-                     dataLength: 512, nullable: true}},
+        db2: {columnName: 'email', dataType: 'VARCHAR',
+          dataLength: 512, nullable: true}},
       name: String,
       bio: Schema.Text,
       birthDate: Date,
       pendingPeriod: Number,
       createdByAdmin: Boolean},
-      {indexes: {index0: {columns: 'email,createdByAdmin'}}}
+    {indexes: {index0: {columns: 'email,createdByAdmin'}}}
```
Comment (dhmlau) [inline test/db2.migration.test.js]: same here.  doesn't seem to need to change the spacing?

**Comment by ssh24** [comment]
@slnode test please

File: test/table.template [hunk 5]
```diff
@@ -0,0 +1,76 @@
+  CREATE TABLE "?"."LOCALUSER1"
+   (    "ID" VARCHAR(20) NOT NULL,
```
Comment (dhmlau) [inline test/table.template]: do we need the quotes around the column names?

File: lib/discovery.js (hunk 2 above)
Comment (ssh24) [inline lib/discovery.js]: The original spacing is incorrect. The new eslint fixed it.

File: test/table.template (hunk 5 above)
Comment (ssh24) [inline test/table.template]: I just took it as it was on the original [tables.sql](https://github.com/strongloop/loopback-connector-db2/blob/master/test/tables.sql) file. So I assume it is needed.

**Comment by ssh24** [comment]
@loay @dhmlau The indenting is correct with the new eslint@8. I updated the eslint version to accommodate string literals (used in pre and post tests) so it can't be in a separate PR either.

**Comment by b-admike** [comment]
@slnode test please

**Comment by ssh24** [comment]
@jannyHou It adds CI for DB2 (docker container) and recover the tests back which were always skipped.

File: test/table.template (hunk 5 above)
Comment (dhmlau) [inline test/table.template]: OK.  I dont' think we need it. but it doesn't hurt to have it since it's coming from the original file.

**Comment by ssh24** [comment]
@slnode test please

**Comment by ssh24** [comment]
@slnode test please

**Comment by ssh24** [comment]
@slnode test please

**Comment by ssh24** [comment]
@slnode test please

**Comment by ssh24** [comment]
@slnode test please

**Pull request merged**